yetanotherco/aligned_layer
# feat: batcher should not be able to build batches too big

# Batcher max_batch_len limit

## Description

This PR adds a feature on the Batcher that prevents it from making batches too big, that would revert in ETH due to gas cost limitations.

## Type of change


## Checklist

  - [Link to the open issue addressing it]() 

## To Test

You can play around with this new `max_batch_len` variable, set in `config-batcher.yaml` .
For example, set it at 16, then send a burst of 20.
You should see the batcher correctly sending one batch of 16 and another of 4.
